dev/io.openliberty.microprofile.health.3.1.internal_fat/fat/src/io/openliberty/microprofile/health31/fat/SlowAppStartupHealthCheckTest.java

```diff
@@ -0,0 +1,274 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.microprofile.health31.fat;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.BufferedReader;
+import java.net.ConnectException;
+import java.net.HttpURLConnection;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Assume;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.HttpUtils;
+
+@RunWith(FATRunner.class)
+public class SlowAppStartupHealthCheckTest {
+
+    private static final String[] EXPECTED_FAILURES = { "CWWKE1102W", "CWWKE1105W", "CWMMH0052W", "CWMMH0054W" };
+
+    public static final String APP_NAME = "DelayedHealthCheckApp";
+    private static final String MESSAGE_LOG = "logs/messages.log";
+
+    private final String STARTED_ENDPOINT = "/health/started";
+
+    private final int SUCCESS_RESPONSE_CODE = 200;
+    private final int FAILED_RESPONSE_CODE = 503; // Response when port is open but Application is not started
+
+    private final String APP_ENDPOINT = "/DelayedHealthCheckApp/DelayedServlet";
+    final static String SERVER_NAME = "SlowAppStartupHealthCheck";
+
+    @Server(SERVER_NAME)
+    public static LibertyServer server1;
+
+    public void setupClass(LibertyServer server, String testName) throws Exception {
+        log("setupClass", testName + " - Deploying the Delayed App into the apps directory and starting the server.");
+
+        WebArchive app = ShrinkHelper.buildDefaultApp(APP_NAME, "io.openliberty.microprofile.health31.delayed.health.check.app");
+        //This test expects to hit the server before the app is started so we disable validation to prevent the test framework waiting for the app to start.
+        ShrinkHelper.exportAppToServer(server, app, DeployOptions.DISABLE_VALIDATION);
+
+        if (!server.isStarted())
+            server.startServer();
+
+        server.waitForStringInLog("CWWKT0016I: Web application available.*DelayedHealthCheckApp*");
```

Comment:
Do we need an assert for CWWKT0016I not found?

Reply:
Updated.